bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java

```diff
@@ -300,28 +374,44 @@ private int handleUpdates(List<Update> updates) {
                 }
 
                 // process metadata
-                lastMessageDate = message.date();
-                lastMessageFirstName = message.from().firstName();
-                lastMessageLastName = message.from().lastName();
-                lastMessageUsername = message.from().username();
-            } else if (update.callbackQuery() != null && update.callbackQuery().message() != null
-                    && update.callbackQuery().message().text() != null) {
-                String[] callbackData = update.callbackQuery().data().split(" ", 2);
+                if (lastMessageURL != null || lastMessageText != null) {
+                    lastMessageDate = message.date();
+                    lastMessageFirstName = message.from().firstName();
+                    lastMessageLastName = message.from().lastName();
+                    lastMessageUsername = message.from().username();
+                }
+            } else if (callbackQuery != null && callbackQuery.message() != null
+                    && callbackQuery.message().text() != null) {
+                String[] callbackData = callbackQuery.data().split(" ", 2);
 
                 if (callbackData.length == 2) {
                     replyId = callbackData[0];
                     lastMessageText = callbackData[1];
-                    lastMessageDate = update.callbackQuery().message().date();
-                    lastMessageFirstName = update.callbackQuery().from().firstName();
-                    lastMessageLastName = update.callbackQuery().from().lastName();
-                    lastMessageUsername = update.callbackQuery().from().username();
-                    chatId = update.callbackQuery().message().chat().id();
-                    replyIdToCallbackId.put(new ReplyKey(chatId, replyId), update.callbackQuery().id());
-                    logger.debug("Received callbackId {} for chatId {} and replyId {}", update.callbackQuery().id(),
-                            chatId, replyId);
+                    lastMessageDate = callbackQuery.message().date();
+                    lastMessageFirstName = callbackQuery.from().firstName();
+                    lastMessageLastName = callbackQuery.from().lastName();
+                    lastMessageUsername = callbackQuery.from().username();
+                    chatId = callbackQuery.message().chat().id();
+                    replyIdToCallbackId.put(new ReplyKey(chatId, replyId), callbackQuery.id());
+
+                    // build and publish callbackEvent trigger channel payload
+                    JsonObject callbackRaw = json.parse(gson.toJson(callbackQuery)).getAsJsonObject();
+                    JsonObject callbackPayload = new JsonObject();
+                    callbackPayload.addProperty("message_id", callbackQuery.message().messageId());
+                    callbackPayload.addProperty("from", String.join(" ",
+                            new String[] { callbackQuery.from().firstName(), callbackQuery.from().lastName() }));
```

Comment:
I was using `String.join()` because it is possible that `lastName` is `null` and for bots `firstName` will also be `null`.

The original code in the binding seems clunky and more (process time) costly than `String.join()`:

```java
updateChannel(LASTMESSAGENAME, (lastMessageFirstName != null || lastMessageLastName != null)
        ? new StringType((lastMessageFirstName != null ? lastMessageFirstName + " " : "")
                + (lastMessageLastName != null ? lastMessageLastName : ""))
        : UnDefType.NULL);
```

I will almalgamte both into a single operation, but what do you suggest between these two:

```java
lastMessageName = (lastMessageFirstName != null || lastMessageLastName != null)
        ? (lastMessageFirstName != null ? lastMessageFirstName : "")
        + (lastMessageLastName != null ? (lastMessageFirstName != null ? " " : "")
        + lastMessageLastName : "") : null;
```

```java
lastMessageName = String.join(" ",
        new String[] { callbackQuery.from().firstName(), callbackQuery.from().lastName() })
```

Reply:
Oh, maybe the second version will throw errors if one of the values is null? I have not tested that specifically and don't know enough about Java to know how it will react.

Reply:
If you want to exclude the nulls I would suggest using a custom utility function to do it. 
```
private static String join(String delimiter, String... elements){
    StringJoiner joiner = new StringJoiner(delimiter);
    for(String element : elements){
        if(element != null){
            joiner.add(element);
        }
    }
    return joiner.toString();
}
```

Reply:
> maybe the second version will throw errors if one of the values is null?

JavaDoc of String.join says:
```
Note that if an element is null, then {@code "null"} is added.
```
The current code therefore does not work as expected.
Best if you go for @cpmeister's suggestion.